packages/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -28,6 +30,25 @@ const GridPaginationRoot = styled(TablePagination)(({ theme }) => ({
   },
 })) as typeof TablePagination;
 
+export type WrappedLabelDisplayedRows = (
+  args: LabelDisplayedRowsArgs & { estimated?: number },
+) => React.ReactNode;
+
+const wrapLabelDisplayedRows = (
+  labelDisplayedRows: WrappedLabelDisplayedRows,
+  estimated?: number,
+): TablePaginationProps['labelDisplayedRows'] => {
+  return ({ from, to, count, page }: LabelDisplayedRowsArgs) =>
+    labelDisplayedRows({ from, to, count, page, estimated });
+};
+
+const defaultLabelDisplayedRows: WrappedLabelDisplayedRows = ({ from, to, count, estimated }) => {
+  if (!estimated) {
+    return `${from}–${to} of ${count !== -1 ? count : `more than ${to}`}`;
+  }
+  return `${from}–${to} of ${count !== -1 ? count : `more than ${estimated > to ? estimated : to}`}`;
```

Comment:
Would it work if instead of wrapping we pass different `to` value depending on situation?
`to: estimated > to ? estimated : to `

Reply:
To achieve something like this, we need at least 3 values, `from`, `to`, and `estimated`, since all of them are different.
![image](https://github.com/mui/mui-x/assets/12609561/c2ad6c18-de74-476f-afd2-2d7a11517ac3)

Does it make sense?

Reply:
Oh, got it.
In that case how about we fork TablePagination translations and move them to our locales? Then we can adapt `labelDisplayedRows` to work with every locale.
What do you think?

Reply:
That's an interesting point.

I initially thought a bit further. i.e. forking the whole `TablePagination` component on our side instead of translations only.
But there were a few reasons why I decided not to go with that approach fat that point.

1. The proper fix would be to do that on the core side since the component belongs in the core. If we move it to the X side, we may miss future updates on the core side (same with the translations)
2. The estimated row count use case doesn't target a huge number of users, we did have a few requests for its support but I doubt that it will be demanded soon by many users, so is the effort even worth it?

I concluded that wrapping could be a nice middle ground. Meanwhile, we could add this functionality on the core side as Flavien [suggested](https://github.com/mui/mui-x/pull/12474#discussion_r1568962914), and when after a major release, we bump the peer dependency on the core we remove this workaround.

Another side of the coin is the material-less grid which we are working on. That makes the argument of completely moving the component (`TablePagination`) on the Grid side a bit stronger, but again there comes a discussion about the implementation of the material-less version since we haven't yet figured out how the isolation between material and material-less versions will work. CC @romgrk (Correct me if I am wrong)

Anyhow, if it looks OK to you, we can proceed with the current workaround for now and settle on something better later?

Reply:
> we could add this functionality on the core side as Flavien https://github.com/mui/mui-x/pull/12474#discussion_r1568962914, and when after a major release, we bump the peer dependency on the core we remove this workaround.

This sounds good to me. I'm not sure bumping the peer dependency on the core is a good idea though, since it would force users to upgrade even if they don't need this. Perhaps a better option would be showing a warning in the docs with the minimum supported version of `@mui/material` required for this to work.

Reply:
> I'm not sure bumping the peer dependency on the core is a good idea though

My proposal is:

1. To implement something on X that do not require bumping the peer dep
2. To have the core support whatever we think is the best DX, if this best DX requires a change on the core
3. To bump the peer dep on the next major (we will probably need it for other topics anyway)